src/common/util/firebase.ts

```diff
@@ -219,3 +225,33 @@ export function indexParsedSentence(sentence: string, tokens: string[], js: stri
     })
     .catch(noop);
 }
+
+export async function getRecentGamesByUserId(userId: string): Promise<Array<object>> {
+  const blueGames = await getRecentGamesForColorByUserId(userId, 'blue');
+  const orangeGames = await getRecentGamesForColorByUserId(userId, 'orange');
+
+  return Object.values({...blueGames, ...orangeGames});
+}
+
+async function getRecentGamesForColorByUserId(userId: string, color: string): Promise<object> {
```

Comment:
as above, I think `object` can be `w.SavedGame`

Reply:
Hmm close. Its a mapping of game IDs to SavedGames. So I think this is the most specific type?

```
Promise<{[gameId: string]: w.SavedGame}>
```